### PR TITLE
docs: Add example of multiplatform add_hook

### PR DESCRIPTION
**Pull request by lukaszkadlubowski**
Adds an example of multi-platform add hook, as discussed in issue #40.

### CODE REVIEW

File: doc/examples.md [hunk 1]
```diff
@@ -196,10 +196,54 @@ else
...
+```bat
+@echo off
+set CHEZMOI_SOURCE_DIR=%userprofile%\.local\share\chezmoi
+python3 "%CHEZMOI_SOURCE_DIR%\.chezmoi_modify_manager.add_hook_wrapped.py" "%1" "%2" "%3"
+```
+
+`.chezmoi_modify_manager.add_hook_wrapped.py` contents:
+
+```python
+# The file from the target directory will be available on STDIN.
```
Comment (VorpalBlade) [inline doc/examples.md:225]:
```suggestion
#!/usr/bin/env python3
# The file from the target directory will be available on STDIN.
```

Needed for everything except Windows.

File: doc/examples.md [hunk 2]
```diff
@@ -196,10 +196,54 @@ else
 fi
 ```
 
-> ⚠️ Windows note! On Windows chezmoi_modify_manager will instead look for a
-file `.chezmoi_modify_manager.add_hook.*`, where `*` is any file extension.
-At most one such file may be present. This allows you to use a suitable
-scripting language for that platform.
+## Add hook on Windows (or multiplatform solution)
+
+On Windows chezmoi_modify_manager will instead look for a file `.chezmoi_modify_manager.add_hook.*`. At most one such file may be present. Currently, `bat` extension is confirmed to be working well. This allows you to use a suitable scripting language for that platform.
```
Comment (VorpalBlade) [inline doc/examples.md:201]:
```suggestion
On Windows chezmoi_modify_manager will instead look for a file `.chezmoi_modify_manager.add_hook.*`. At most one such file may be present. This allows you to use a suitable scripting language for that platform. Currently, `bat` extension is confirmed to be working well. However not all scripting languages seem to work here (for unknown reason).
```

**Review by VorpalBlade — CHANGES_REQUESTED** [review]
Broadly good, with some minor tweaks.

Excellent grammar and spelling (from what I can tell as a non-native speaker).

**Comment by VorpalBlade** [comment]
Do you know if there is any possibility of doing something equivalent to `CHEZMOI_SOURCE_DIR="$(chezmoi source-path)"` in the bat file? If not, that is fine, but it would be nice.

**Comment by lukaszkadlubowski** [comment]
> Do you know if there is any possibility of doing something equivalent to `CHEZMOI_SOURCE_DIR="$(chezmoi source-path)"` in the bat file? If not, that is fine, but it would be nice.

Yes, I'm working on it now. Sorry, I didn't know how to postpone yet another pull request until I pull changes and test them locally. I'll let you know once it is ready.

**Comment by lukaszkadlubowski** [comment]
Ok, I added commit with bat file modification. Thank you for the feedback!

**Comment by VorpalBlade** [comment]
```bat
for /f %%i in ('chezmoi source-path') do set "CHEZMOI_SOURCE_DIR=%%i"
```

Bat files really are a terrible language. Comparatively Powershell seems positively sane.

**Review by VorpalBlade — APPROVED** [review]

**Comment by lukaszkadlubowski** [comment]
> ```batchfile
> for /f %%i in ('chezmoi source-path') do set "CHEZMOI_SOURCE_DIR=%%i"
> ```
> 
> Bat files really are a terrible language. Comparatively Powershell seems positively sane.

Yes, bat is an ancient era, but I think that Microsoft has done a really good job with PowerShell in terms of syntax consistency and code readability. PowerShell 7 is also multiplatform. I have never tested it under other OS, but sounds promising.

**Comment by VorpalBlade** [comment]
I looked at powershell early on (whatever the first public version was), and it seemed to be an object oriented mess with very long command names. *Might* be ok for scripting, definitely not for interactive use. And by that point I was already firmly a Linux-first user, so it never made sense to look further at it.

**Pull request merged**